src/client/client.cpp

```diff
@@ -309,6 +312,7 @@ void Client::Stop()
 	if (m_localdb) {
 		infostream << "Local map saving ended." << std::endl;
 		m_localdb->endSave();
+		m_localdb.reset();
```

Comment:
Why? Just let it go out of scope when the `~Client` is called. `Stop` is only ever called before the client is destroyed.
This adds risk without any clear advantage.